### PR TITLE
[MST-535] Prevent IDV name change if name is managed by a third party

### PR DESCRIPTION
**Pull request by bseverino**
[MST-535](https://openedx.atlassian.net/browse/MST-535)

- Currently, if a learner's profile fields are locked due to SSO, and they attempt to change their name during the IDV flow, they will encounter an error upon submitting. This change implements several warnings for these learners before IDV submission, and prevents them from attempting to change their name manually.
- Also fixed a bug where the name change field was resetting to to original name when cleared.

---
Screen Shot 2021-03-15 at 11 17 39 AM
---
Screen Shot 2021-03-15 at 11 18 10 AM
---
Screen Shot 2021-03-15 at 11 18 38 AM

### CODE REVIEW

File: src/id-verification/IdVerificationContextProvider.jsx [hunk 1]
```diff
@@ -2,24 +2,74 @@ import React, { useState, useContext, useEffect } from 'react';
...
+
+import { getExistingIdVerification, getEnrollments } from './data/service';
 import AccessBlocked from './AccessBlocked';
+import { hasGetUserMediaSupport } from './getUserMediaShim';
 import IdVerificationContext, { MEDIA_ACCESS, ERROR_REASONS, VERIFIED_MODES } from './IdVerificationContext';
 
 export default function IdVerificationContextProvider({ children }) {
+  const { authenticatedUser } = useContext(AppContext);
+
   const [existingIdVerification, setExistingIdVerification] = useState(null);
```
Comment (bseverino) [inline src/id-verification/IdVerificationContextProvider.jsx:16]: Grouped together related logic in this file for better readability.

File: src/id-verification/panels/GetNameIdPanel.jsx [hunk 2]
```diff
@@ -45,6 +46,34 @@ function GetNameIdPanel(props) {
...
+    }
+    return nameOnAccountValue;
+  };
+
+  const getErrorMessage = () => {
+    if (profileDataManager) {
+      return (
+        <FormattedMessage
+          id="id.verification.account.name.managed.alert"
+          defaultMessage="Your name is managed by {managerTitle}. Please contact your administrator or {support} for help."
```
Comment (schenedx) [inline src/id-verification/panels/GetNameIdPanel.jsx:61]: I feel like this message is not detailed enough. 
What about "We have detected that you are a learner affiliated with {managerTitle}. You are not allowed to update this name. Please contact system administrator of {managerTitle} for help"?

**Review by schenedx — COMMENTED** [review]
See message update recommendation.

I'd love to see a demo of this.

File: src/id-verification/panels/GetNameIdPanel.jsx (hunk 2 above)
Comment (bseverino) [reply src/id-verification/panels/GetNameIdPanel.jsx:61]: @schenedx Before I push the changes again - how does this sound? "Your profile settings are managed by {managerTitle}, so you are not allowed to update this name. Please contact your {managerTitle} administrator or {support} for help."

Comment (schenedx) [reply src/id-verification/panels/GetNameIdPanel.jsx:61]: Sounds good to me.

File: src/id-verification/panels/GetNameIdPanel.jsx [hunk 3]
```diff
@@ -45,6 +46,39 @@ function GetNameIdPanel(props) {
     }
   }, [nameMatches, blankName]);
 
+  const getNameValue = () => {
+    if (!nameMatches) {
+      // idPhotoName should also be used if it's an empty string, otherwise
+      // it will reset to nameOnAccountValue when the field is cleared
+      if (typeof idPhotoName === 'string') {
```
Comment (schenedx) [inline src/id-verification/panels/GetNameIdPanel.jsx:53]: #NIT can you comment on here why you are checking the type of idPhotoName? Are you just preventing this field to be null?

**Review by schenedx — APPROVED** [review]
👍 LGTM

File: src/id-verification/panels/GetNameIdPanel.jsx (hunk 3 above)
Comment (bseverino) [reply src/id-verification/panels/GetNameIdPanel.jsx:53]: If I just check that idPhotoName exists, an empty string will return falsy (which was causing it to reset before)

Comment (schenedx) [reply src/id-verification/panels/GetNameIdPanel.jsx:53]: Yep. I understand that. I was suggesting putting that explanation in the comments in the code so others understand what you are doing here.

Comment (bseverino) [reply src/id-verification/panels/GetNameIdPanel.jsx:53]: Got it, I'll make the comment a little clearer

**Pull request merged**